firmware/Sources/MID/MID_DABS/mid_dabs.c

```diff
@@ -0,0 +1,292 @@
+};
+const blink_conf_s cvDchgMode = {
+	blink_mode_cv_dchg, 5,	{0x0F,0x07,0x0B,0x0D,0x0E}
+};
+const blink_conf_s cpChgMode = {
+	blink_mode_cp_chg, 5,	{0x00,0x01,0x03,0x07,0x0F}
+};
+const blink_conf_s cpDchgMode = {
+	blink_mode_cp_dchg, 5,	{0x00,0x08,0x0C,0x0E,0x0F}
+};
+const blink_conf_s errHsvoltMode = {
+	blink_mode_err_hsvolt, 1,	{0x01,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errlsvoltMode = {
+	blink_mode_err_lsvolt, 1,	{0x08,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errlscurrMode = {
+	blink_mode_err_lscurr, 1, {0xC,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errCommMode = {
+	blink_mode_err_comm, 1,	{0x0A,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errTempMode = {
+	blink_mode_err_temp, 1,	{0x03,0x00,0x00,0x00,0x00}
+};
+const blink_conf_s errIntMode = {
+	blink_mode_err_int, 1,	{0x09,0x00,0x00,0x00,0x00}
+};
+/**********************************************************************************/
+/*                         Definition of local variables                          */
+/**********************************************************************************/
+blink_s ledsMode = {
+		idleMode, 0, idleMode //Initialice in idle mode
+};
+/**********************************************************************************/
+/*                        Definition of exported variables                        */
+/**********************************************************************************/
+/**********************************************************************************/
+/*                        Definition of imported variables                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local functions                          */
+/**********************************************************************************/
+/**
+ * @fn MID_DABS_result_e CheckBlinkStatus(MID_REG_mode_e epcmode, int16_t lscurr,
+		MID_REG_errorStatus_s errors, blink_conf_s * mode)
+ * @brief Get the leds mode associated to the mode in which is currently working the epc.
+ *
+ * @param epcmode MID_REG_mode_e, is the mode the epc is working
+ * @param lscurr, is the actual measurement of the current in the low side,
+ * 			depending on the value will be charging or discharging.
+ * @param errors, give access to the errors register to now if there are any error.
+ * @param mode, Output of the function to give the mode for the leds.
+ * @return @ref MID_DABS_RESULT_SUCCESS if all data gathered correctly,
+ * 		@ref MID_DABS_RESULT_BUSY, @ref MID_DABS_RESULT_TIMEOUT or
+ * 		@ref MID_DABS_RESULT_ERROR otherwise.
+ */
+static MID_DABS_result_e CheckBlinkStatus(MID_REG_mode_e epcmode, int16_t lscurr,
+		MID_REG_errorStatus_s * errors, blink_conf_s * mode){
+	MID_DABS_result_e res = MID_DABS_RESULT_SUCCESS;
+	if (errors->hsVoltErr !=MID_REG_ERROR_NONE){
```

Comment:
we have to check priority

Reply:
As we discuss the other day, the final priority is Internal, HS volt, communications, LS volt, LS curr and Temperatures.